core/src/main/java/org/eclipse/hono/connection/ConnectionFactoryImpl.java

```diff
@@ -38,8 +38,22 @@
 public class ConnectionFactoryImpl implements ConnectionFactory {
 
     private static final Logger logger = LoggerFactory.getLogger(ConnectionFactoryImpl.class);
-    private Vertx                      vertx;
-    private ClientConfigProperties config;
+    private Vertx vertx;
+    private final ClientConfigProperties config;
+
+    /**
+     * Constructor with the configuration parameters for connecting to the 
+     * AMQP server as parameter.
+     * <p>
+     * The <em>name</em> property of the configuration is used as the basis
+     * for the local container name which is then appended with a UUID.
+     * 
+     * @param config The configuration parameters.
+     * @throws NullPointerException if the parameters are {@code null}.
+     */
+    public ConnectionFactoryImpl(final ClientConfigProperties config) {
```

Comment:
if we replace the setter for the config with a constructor param then IMHO we should do the same for the vertx instance as well for consistency reasons, shouldn't we?